src/pages/periodValidity.tsx

```diff
@@ -1,30 +1,84 @@
 import CsrfForm from '../components/CsrfForm';
 import { getSessionAttribute } from '../utils/sessions';
-import { PeriodExpiryWithErrors } from './api/periodValidity';
-import { getCsrfToken } from '../utils';
+import { getCsrfToken, getErrorsByIds } from '../utils';
+import RadioConditionalInput, { RadioConditionalInputFieldset } from '../components/RadioConditionalInput';
+import { isWithErrors } from '../interfaces/typeGuards';
 
 const title = 'Period Validity - Create Fares Data Service';
 const description = 'Period Validity selection page of the Create Fares Data Service';
 
-const errorId = 'period-end-calendar';
-
 type PeriodValidityProps = {
     errors?: ErrorInfo[];
+    fieldsets: RadioConditionalInputFieldset[];
     csrfToken: string;
 };
 
-const isPeriodExpiryWithErrors = (
-    periodExpiryAttribute: ProductData | PeriodExpiryWithErrors,
-): periodExpiryAttribute is PeriodExpiryWithErrors =>
-    (periodExpiryAttribute as PeriodExpiryWithErrors)?.errorMessage !== undefined;
+export const getFieldsets = (errors: ErrorInfo[]): RadioConditionalInputFieldset[] => {
+    const periodValidityFieldSet: RadioConditionalInputFieldset = {
+        heading: {
+            id: 'period-validity',
+            content: 'Is this ticket only valid on certain days or times?',
+            hidden: true,
+        },
+        radios: [
+            {
+                id: 'period-end-calendar',
+                name: 'periodValid',
+                value: 'endOfCalendarDay',
+                label: ' At the end of a calendar day',
+                radioButtonHint: {
+                    id: 'period-end-calendar-hint',
+                    content:
+                        ' For example, a ticket purchased at 3pm would be valid until midnight on its day of expiry',
+                },
+            },
+            {
+                id: 'period-twenty-four-hours',
+                name: 'periodValid',
+                value: '24hr',
+                label: 'At the end of a 24 hour period from purchase',
+                radioButtonHint: {
+                    id: 'period-twenty-four-hours-hint',
+                    content: 'For example, a ticket purchased at 3pm will be valid until 3pm on its day of expiry',
+                },
+            },
+            {
+                id: 'period-end-of-service',
+                name: 'periodValid',
+                value: 'endOfServiceDay',
+                dataAriaControls: 'period-validity-end-of-service-required-conditional',
+                label: 'End of service day',
+                radioButtonHint: {
+                    id: 'period-end-of-service-hint',
+                    content:
+                        'For example, a ticket purchased at 3pm would be valid until the end of your service day on its day of expiry',
+                },
+                inputHint: {
+                    id: 'end-of-service-day-hint',
+                    content: '',
```

Comment:
I wonder if we do want to have some content on this hint? We could pass a `hidden` flag through to not have it rendered? Screen readers might get confused if there is a hint without any content.

Reply:
tried to do it with content being empty but did not work. Think it needs a refactor really so we dont have to pass it through.